### PR TITLE
[WIP] Implement `gettxout` RPC

### PR DESCRIPTION
**Pull request by nopara73**
I added tests, too.  
This PR is in WIP status, because I am not able to run the tests, as I explained here: https://github.com/MetacoSA/NBitcoin/pull/312

### CODE REVIEW

**Comment by nopara73** [comment]
It seems like appveyor test will fail, because RPC command returns null, which means the txout is not unspent.  

Does appveyor run the tests on bitcoin core testnet?

File: NBitcoin.Tests/RPCClientTests.cs [hunk 1]
```diff
@@ -161,6 +161,26 @@ public void CanGetBlockFromRPC()
...
 
+		[Fact]
+		public async Task CanGetUtxoFromRPCAsync()
+		{
+			using (var builder = NodeBuilder.Create())
+			{
+				var rpc = builder.CreateNode().CreateRPCClient();
+				builder.StartAll();
+				// nopara73 burned these coins
+				GetTxOutResponse getTxOutResponse = await rpc.GetUnspentTransactionOutputAsync(new uint256("eed54c03582962b251f561b45213ce12af2a25228bfa8ca646bdb3c9b80e1079"), 1, false);
```
Comment (NicolasDorier) [inline NBitcoin.Tests/RPCClientTests.cs]: Can you call it `GetTxout` to stay near from bitcoin naming?

**Comment by NicolasDorier** [comment]
It runs tests on regtest. Just mine 101 blocks with `generate(101)`

**Comment by nopara73** [comment]
@NicolasDorier I've finished your asked modifications and, I created a comprehensive test.

**Comment by NicolasDorier** [comment]
A conflict caused by another of your PR

**Comment by NicolasDorier** [comment]
Do you have more stuff in the pipeline? after this PR, and once I fixed the build on appveyor, will release a new package or better to wait?

**Comment by nopara73** [comment]
> Do you have more stuff in the pipeline?

I just had some free time yesterday. Today I'm concentrating on other things.

**Comment by nopara73** [comment]
@NicolasDorier, appveyor seems to be complaining about something. I didn't have time to take a look at it yet, quick skimming doesn't help. Are you sure it's ok?

**Comment by NicolasDorier** [comment]
yeah I am checking now

**Comment by nopara73** [comment]
Fixed here: https://github.com/MetacoSA/NBitcoin/commit/604ae7a70b76df85769cc3500600b48ceb57aa2d

**Pull request merged**